sdk/formrecognizer/ai-form-recognizer/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,125 @@
+- [Logging](#logging)
+
+## Troubleshooting Errors
+
+### Handling RestError
+
+Form Recognizer service methods throw [`RestError`] on failure.
+
+The RestError raised by the Azure Form Recognizer client library includes detailed error response information that provides useful insights into what went wrong and includes corrective actions to fix common issues.
+This error information can be found in the `RestError#message` property.
+
+### Build model error
+
+Build model errors usually occur when trying to build a custom model. The most common scenarios when this might occur is, if you are building the model with an
+[Invalid data set](#invalid-training-data-set) or an [Invalid SAS Url](#invalid-sas-url).
+
+#### Invalid training data set
+
+This error indicates that the provided data set does not match the training data requirements.
+Learn more about building a training data set, [here](https://aka.ms/customModelV3).
+
+Example error below:
+
+```bash
+RestError: Invalid argument.
+  {
+    "name": "RestError",
+    "code": "InvalidArgument",
+    "statusCode": 400,
+    .
+    .
+    .
+    "details": {
+     "error": {
+       "code": "InvalidArgument",
+       "message": "Invalid argument.",
+       "innererror": {
+         "code": "InvalidParameter",
+         "message": "The parameter ContainerUrl is invalid: Url is null or whitespace"
+       }
+     }
+    }
+  }
+```
+
+#### Invalid SAS URL
+
+This error points to missing permissions on the blob storage SAS URL for the Form Recognizer service to access the training dataset resource. For more information about SAS tokens for Form Recognizer, see [here](https://learn.microsoft.com/azure/applied-ai-services/form-recognizer/create-sas-tokens).
+
+Example error:
+
+```bash
+ RestError: Invalid argument.
+  {
+   "name": "RestError",
+   "code": "InvalidArgument",
+   "statusCode": 400,
+   .
+   .
+   .
+   "details": {
+     "error": {
+       "code": "InvalidArgument",
+       "message": "Invalid argument.",
+       "innererror": {
+         "code": "InvalidSasToken",
+         "message": "The shared access signature (SAS) is invalid: SAS token expired on 10/04/2022 03:19:05 +00:00"
+       }
+     }
+   },
+   "message": "Invalid argument."
+ }
+```
+
+### Generic Error
+
+Seeing a "Generic error" returned from the SDK is most often caused by heavy load on the service. For troubleshooting issues related to service limits, see related information [here](https://learn.microsoft.com/azure/applied-ai-services/form-recognizer/service-limits?tabs=v30).
+
+Example error:
+
+```bash
+(3014) Generic error during training.
+Invalid model created with ID=<model ID>
+```
+
+### Unexpected time to build a custom model
+
+It is common to have a longer completion time than what is expected when building a custom model using the `neural` build mode with `DocumentModelAdministrationClient#beginBuildDocumentModel()`. Depending on the service load you can usually expect it to take ~10min.
```

Comment:
"Longer completion time than expected" doesn't feel quite right, because we do expect the neural build mode to take a long time. I would rephrase as "Training with the "neural" build mode typically takes significantly longer than the "template" mode.

Reply:
Copied this from Python/Java guides.

cc @catalinaperalta @samvaity